src/fbx/Fbx2Raw.cpp

```diff
@@ -367,13 +367,60 @@ static void ReadMesh(RawModel &raw, FbxScene *pScene, FbxNode *pNode, const std:
     }
 }
 
+//ar : aspectY / aspectX
+double HFOV2VFOV(double h, double ar)
+{
+    return 2.0 * std::atan((ar) * std::tan((h * FBXSDK_PI_DIV_180) * 0.5)) * FBXSDK_180_DIV_PI;
+};
+
+//ar : aspectX / aspectY
+double VFOV2HFOV(double v, double ar)
+{
+    return 2.0 * std::atan((ar) * std::tan((v * FBXSDK_PI_DIV_180) * 0.5)) * FBXSDK_180_DIV_PI;
+}
+
+// Largely adopted from fbx example 
 static void ReadCamera(RawModel &raw, FbxScene *pScene, FbxNode *pNode)
 {
     const FbxCamera *pCamera = pNode->GetCamera();
+
+    double filmHeight = pCamera->GetApertureHeight();
+    double filmWidth = pCamera->GetApertureWidth() * pCamera->GetSqueezeRatio();
+    
+    // note Height : Width
+    double apertureRatio = filmHeight / filmWidth;
+
+    double fovx = 0.0f;
+    double fovy = 0.0f;
+
+    switch(pCamera->GetApertureMode())
+    {
+        case FbxCamera::EApertureMode::eHorizAndVert: {
+            fovx = pCamera->FieldOfViewX;
+            fovy = pCamera->FieldOfViewY;
+            break;
+        }
+        case FbxCamera::EApertureMode::eHorizontal: {
+            fovx = pCamera->FieldOfViewX;
+            fovy = HFOV2VFOV(fovx, apertureRatio);
+            break;
+        }
+        case FbxCamera::EApertureMode::eVertical: {
+            fovy = pCamera->FieldOfViewY;
+            fovx = VFOV2HFOV(fovy, 1.0 / apertureRatio);
+            break;
+        }
+        case FbxCamera::EApertureMode::eFocalLength: {
+            fovx = pCamera->ComputeFieldOfView(pCamera->FocalLength);
+            fovy = HFOV2VFOV(fovx, apertureRatio);
+            break;
+        }
+    }
```

Comment:
Would perhaps make sense to add a default case where you set the fovx/y to pCamera->FieldOfViewX/Y. Optionally I guess there should be a default case where it throws or similar. Carrying on with fov = 0 seems bad.

Reply:
The switch covers all possible EApertureMode values in current sdk; we could add some default case to try and future proof it, but pCamera->FieldOfViewX/Y and FocalLength have non-zero default values in that case. So then you get weird values in the resulting glTF file and it is hard to figure out where things went wrong (this is how master behaves currently).

Would you prefer default case now knowing it is exhaustive switch? I don't mind either way, so your call.

Reply:
I see. In that case, just to make it easier to debug future issues, we could log a warning/error in the default case
fmt::printf("Warning:: Unsupported ApertureMode");

How about that?